lib/phoenix/socket/router.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Phoenix.Socket.Router do
+  # Routes WebSocket and LongPoller requests.
+  @moduledoc false
+
+  use Plug.Builder
+  alias Phoenix.Transports.WebSocket
+  alias Phoenix.Transports.LongPoller
+
+  plug Plug.Logger
+  plug :fetch_query_params
+  plug :dispatch
+
+  def dispatch(conn, _) do
+    case {conn.method, transport(conn)} do
```

Comment:
You know I rewrote this about six times with and without multiple dispatch heads. I generally prefer multiple heads, so may change :)